packages/main/src/Table.ts

```diff
@@ -464,11 +464,17 @@ class Table extends UI5Element {
 	}
 
 	get styles() {
-		return {
+		const tableStyle = {
 			table: {
 				"grid-template-columns": this._gridTemplateColumns,
 			},
-		};
+		} as any;
+
+		this.headerRow[0].cells.forEach(headerCell => {
+			tableStyle.table[`--h-align-${(headerCell as any)._individualSlot}`] = headerCell.hAlign;
+		});
+
+		return tableStyle;
```

Comment:
Comes down to personal preference, but I'd like to keep returning the object instance directly. You could use reduce and the spread operator to achieve this:

```ts
const headerStyleMap = this.headerRow[0].cells.forEach((headerStyles, headerCell) => {
  headerStyles[`--h-align-${(headerCell as any)._individualSlot}`] = headerCell.hAlign;
  return headerStyles;
}, {} as any);

return {
  table: {
    "grid-template-columns": this._gridTemplateColumns,
    ...headerStyleMap
  }
};
```

Again this boils down to personal preference. I am also fine with keeping it as is.

Reply:
Should also solve your linter issue:

 |   477:3  error  Unsafe return of an `any` typed value  @typescript-eslint/no-unsafe-return

Seems like there is a linter rule, that any types should not be returned (which kinda makes sense)

Reply:
As discussed I restrict the values of hAlign to the ones in TableCellHorizontalAlign.

Reply:
I've added _individualSlot as a member to TableHeaderCell.